src/main/java/dan200/computercraft/core/apis/SoundAPI.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * This file is part of ComputerCraft - http://www.computercraft.info
+ * Copyright Daniel Ratcliffe, 2011-2016. Do not distribute without permission.
+ * Send enquiries to dratcliffe@gmail.com
+ *
+ */
+
+package dan200.computercraft.core.apis;
+
+import dan200.computercraft.ComputerCraft;
+import dan200.computercraft.api.lua.LuaException;
+import dan200.computercraft.api.lua.ILuaContext;
+import dan200.computercraft.shared.computer.core.ServerComputer;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.SoundEvent;
+
+public class SoundAPI implements ILuaAPI
+{
+
+    /**
+     * Sound API for ComputerCraft. Provides an interface to Forge's API.
+     * Possible use cases are notification apis, alert sounds, or playing music
+     */
+
+    private String[] methods;
+    private ServerComputer computer;
+    private long lastPlayTime;
+
+    public SoundAPI(ServerComputer computer)
+    {
+
+        this.methods = new String[] {
+                "play", // Plays a sound from given resource locator
+                        // Returns: Object[]{success}
+                "getPlayTimeout" // Gets minTimeBetweenPlay
+        };
+
+        this.computer = computer;
+        this.lastPlayTime = 0;
+
+    }
+
+
+    /* ILuaAPI implementations */
+
+    @Override
+    public String[] getNames()
+    {
+        return new String[] {
+                "sound",
+        };
+    }
+
+    @Override
+    public String[] getMethodNames()
+    {
+        return this.methods;
+    }
+
+    @Override
+    public void startup()
+    {
+    }
+
+    @Override
+    public void advance(double _dt)
+    {
+    }
+
+    @Override
+    public void shutdown()
+    {
+
+    }
+
+    @Override
+    public Object[] callMethod(ILuaContext context, int method, Object[] arguments) throws LuaException, InterruptedException
+    {
+        switch (method) {
+
+            // play
+            case 0:
+
```

Comment:
Wrap each case statement in braces, so variables are not shared between them. It isn't strictly needed now, but if additional methods are added you'll want them.